src/app/src/components/Submissions/Detail/Map.js

```diff
@@ -19,40 +30,103 @@ export default function Map({ submission }) {
             >
                 <MapPanes>
                     <DefaultBasemap />
-                    <MapButtons submission={submission} />
+                    <MapButtons
+                        id={id}
+                        status={status}
+                        email={submission.primary_contact.email}
+                        shape={submission.shape}
+                    />
+                    <Polygon shape={submission.shape} />
                 </MapPanes>
             </MapContainer>
-            <SubmissionStatusBar submission={submission} />
+            <SubmissionStatusBar status={status} />
         </>
     );
 }
 
-function MapButtons({ submission }) {
+function MapButtons({ id, status, email, shape }) {
+    const navigate = useNavigate();
+    const userRole = useSelector(state => state.auth.user.role);
+
+    const canEditDraft =
+        userRole === ROLES.CONTRIBUTOR || userRole === ROLES.ADMINISTRATOR;
+    const canReview =
+        userRole === ROLES.VALIDATOR || userRole === ROLES.ADMINISTRATOR;
+
+    const getDrawButtonText = () => {
+        switch (status) {
+            case BOUNDARY_STATUS.DRAFT:
+                return canEditDraft ? 'Edit boundary' : null;
+            case BOUNDARY_STATUS.SUBMITTED:
+                return canReview ? 'Start review' : null;
+            case BOUNDARY_STATUS.IN_REVIEW:
+                return canReview ? 'Continue review' : null;
+            default:
+                return null;
+        }
+    };
+
+    const drawButtonText = getDrawButtonText();
+
     return (
         <Box position='absolute' zIndex={1000} top={22} w='100%'>
             <Flex justify='space-evenly'>
-                <MapButton icon={ChatAltIcon}>Review map</MapButton>
-                <MapButton icon={MailIcon}>Email</MapButton>
-                <MapButton icon={DownloadIcon}>Download</MapButton>
+                {drawButtonText ? (
+                    <MapButton
+                        icon={ChatAltIcon}
+                        onClick={() => navigate(`/draw/${id}/`)}
+                    >
+                        {drawButtonText}
+                    </MapButton>
+                ) : null}
+                <MapButton icon={MailIcon}>
+                    <a href={`mailto:${email}`} style={{ color: 'inherit' }}>
+                        Email
+                    </a>
+                </MapButton>
+                <MapButton
+                    icon={DownloadIcon}
+                    onClick={() =>
+                        downloadData(JSON.stringify(shape), 'boundary.geojson')
```

Comment:
Consider adding a utility function that takes a boundary and returns the download filename.

It should be of the format `YYYY_MM_DD_UTILITY_NAME.geojson`, where the date corresponds to the date of submission, and the utility name is uppercased and snakecased.

Use that function here.